### ISSUE TITLE
Investigate user logging out on page reload

### ISSUE THREAD

**Issue by punchagan** [india-ultimate/hub#206]
Some users seem to be getting logged out on page reloads (on mobile?). This would be something good to investigate and fix.

It could be that the `/me` API request takes a while to return and the user feels they are logged out, or some such thing.

**Comment by punchagan** [comment]
I'm not sure why so many people would notice it, then. The link would be hidden by default. Is something else making them check the link? : thinking:

**Comment by Joe2k** [comment]
@punchagan So this we can check in clarity for user behavior for deep understanding. But what I observed is people dont know that clicking breadcrumbs would take them back.. Instead try to click on Tournaments in Navbar.. Hence they open navbar and see Login button and think they are logged out

**Comment by Joe2k** [comment]
One of feedback in related context:
"A profile picture or a logged in option at te top wold be good to indicate if i have logged in or not."

**Comment by punchagan** [comment]
> Hence they open navbar and see Login button and think they are logged out

Maybe it's with fixing that bug -- "login" being shown when the user is logged in.

**Comment by punchagan** [comment]
Ah, I see what's happening here. Since the tournament pages are available without logging in, we aren't fetching the user info on reload and the Login option is shown when the store doesn't contain user information.

**Comment by SibiAkkash** [comment]
After a reload, clicking on Dashboard seems to bring back the logged in state.
https://github.com/india-ultimate/hub/assets/39093537/2b6b6f88-41fd-4009-8a7d-21e09d56dfdb

There was a bug while clicking the breadcrumbs, which i've fixed. Still understanding why clicking the dashboard brings back the state

**Comment by punchagan** [comment]
Yes, that's correct. Sorry for the cryptic and short note, but yes, that's what I was implying in the note above. Thanks Sibi!

**Comment by SibiAkkash** [comment]
> Yes, that's correct. Sorry for the cryptic and short note, but yes, that's what I was implying in the note above. Thanks Sibi!

No problem, I didn't think much about your note till I looked at App.js, so I just wrote down what I understood 😅

**Comment by SibiAkkash** [comment]
Closed by https://github.com/india-ultimate/hub/commit/28821d77c730af211b510c07f2efe58f41f13df1
